neps/nep-0488.md

```diff
@@ -0,0 +1,1212 @@
+
+$$
+F_{p^2} = F_p[u] / (u^2 + 1)
+$$
+
+$$
+F_{p^6} = F_{p^2}[v] / (v^3 - u - 1)
+$$
+
+$$
+F_{p^{12}} = F_{p^6}[w] / (w^2 - v)
+$$
+
+Generator for $G_1$:
+
+- $x = \mathtt{0x17f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb}$
+- $y = \mathtt{0x08b3f481e3aaa0f1a09e30ed741d8ae4fcf5e095d5d00af600db18cb2c04b3edd03cc744a2888ae40caa232946c5e7e1}$
+
+Generator for $G_2$:
+
+- $x_0 = \mathtt{0x024aa2b2f08f0a91260805272dc51051c6e47ad4fa403b02b4510b647ae3d1770bac0326a805bbefd48056c8c121bdb8}$
+- $x_1 = \mathtt{0x13e02b6052719f607dacd3a088274f65596bd0d09920b61ab5da61bbdc7f5049334cf11213945d57e5ac7d055d042b7e}$
+- $y_0 = \mathtt{0x0ce5d527727d6e118cc9cdc6da2e351aadfd9baa8cbdd3a76d429a695160d12c923ac9cc3baca289e193548608b82801}$
+- $y_1 = \mathtt{0x0606c4a02ea734cc32acd2b02bc28b99cb3e287e85a763af267492ab572e99ab3f370d275cec1da1aaa9075ff05f79be}$
+
+
+Cofactor for $G_1$:
+$$h = \mathtt{0x396c8c005555e1568c00aaab0000aaab}$$
+
+Cofactor for $G_2$:
+$$h' = \mathtt{0x5d543a95414e7f1091d50792876a202cd91de4547085abaa68a205b2e5a7ddfa628f1cb4d9e82ef21537e293a6691ae1616ec6e786f0c70cf1c38e31c7238e5}$$
+
+Key  BLS12-381 parameter used in Miller Loop:
+$$x = -\mathtt{0xd201000000010000}$$
+
+All parameters were sourced from [^15], [^51], and [^14], and they remain consistent across these sources.
+
+### Map to curve specification
+
+This section delineates the functionality of the `bls12381_map_fp_to_g1` and `bls12381_map_fp2_to_g2` functions, 
+operating in accordance with the RFC9380 specification "Hashing to Elliptic Curves"[^62].
+
+These functions map field elements in $F_p$ or $F_{p^2}$ 
+to their corresponding subgroups: $G_1 \subset E(F_p)$ or $G_2 \subset E'(F_{p^2})$. 
+`bls12381_map_fp_to_g1`/`bls12381_map_fp2_to_g2` combine the functionalities 
+of `map_to_curve` and `clear_cofactor` from RFC9380[^63].
+
+```text
+fn bls12381_map_fp_to_g1(u):
+    let Q = map_to_curve(u);
+    return clear_cofactor(Q);
+```
+
+We choose not to implement the `hash_to_field` function as a host function due to potential changes in hashing methods. 
+Additionally, executing this function within the contract consumes approximately 2 TGas, which is acceptable for our goals.
+
+Specific implementation parameters for `bls12381_map_fp_to_g1` and `bls12381_map_fp2_to_g2` can be found in RFC9380 
+under sections 8.8.1[^64] and 8.8.2[^65], respectively.
+
+### Curve points encoding
+
+#### General comments
+
+The encoding rules for curve points and field elements align with the standards established in zkcrypto[^53] and 
+the implementation in the milagro lib[^29].
+
+For elements from $F_p$ the first three bits will always be $0$, because the first byte of $p$ equals $1$. As a result, 
+we can use these bits to encode extra information: the encoding format, the point at infinity, and the points' sign. 
+Read more in sections: Uncompressed/compressed points on curve $E(F_p)$/$E'(F_{p^2})$.
```

Comment:
GitHub doesn't render the latter LaTeX expression `$E'(F_{p^2})$` .

Reply:
fixed: https://github.com/near/NEPs/pull/488/commits/52f95599df93c2608e1a7304a5b4bc54de5fcaa5